packages/services/api/src/modules/organization/providers/organization-access-tokens.ts

```diff
@@ -315,31 +567,533 @@ export class OrganizationAccessTokens {
+    });
+
+    const accessToken = await this.getById(id);
+
+    if (!accessToken || accessToken?.organizationId !== organization.id) {
       return null;
     }
 
-    return OrganizationAccessTokenModel.parse(row);
+    return accessToken;
+  }
+
+  async getForMembership(membership: OrganizationMembership, id: string) {
+    const accessToken = await this.getById(id);
+
+    if (
+      !accessToken ||
+      !accessToken.userId ||
+      accessToken.userId !== membership.userId ||
+      accessToken.organizationId !== membership.organizationId
+    ) {
+      return null;
+    }
+
+    return accessToken;
+  }
+
+  async getForProject(project: Project, id: string) {
+    await this.session.assertPerformAction({
+      organizationId: project.orgId,
+      params: { organizationId: project.orgId, projectId: project.id },
+      action: 'projectAccessToken:modify',
+    });
+
+    const accessToken = await this.getById(id);
+
+    if (!accessToken || accessToken?.projectId !== project.id) {
+      return null;
+    }
+
+    return accessToken;
+  }
+
+  async getAvailablePermissionsGroupsForProject(project: Project): Promise<Array<PermissionGroup>> {
+    await this.session.assertPerformAction({
+      organizationId: project.orgId,
+      params: { organizationId: project.orgId, projectId: project.id },
+      action: 'projectAccessToken:modify',
+    });
+
+    const organization = await this.storage.getOrganization({ organizationId: project.orgId });
+    const groups = await this.getAvailablePermissionGroupsForOrganization(organization);
+
+    return groups
+      .map(group => ({
+        ...group,
+        permissions: group.permissions.filter(permission =>
+          validProjectResourceLevels.has(getPermissionGroup(permission.id)),
+        ),
+      }))
+      .filter(group => group.permissions.length !== 0);
+  }
+
+  private createFeatureFlagPermissionFilter(organization: Organization) {
+    const isAppDeplymentsEnabled = organization.featureFlags.appDeployments;
+    const isOTELTracingEnabled = organization.featureFlags.otelTracing;
+
+    return (id: Permission) =>
+      (!isAppDeplymentsEnabled && id.startsWith('appDeployment:')) ||
+      (!isOTELTracingEnabled && id.startsWith('traces:'))
+        ? false
+        : true;
+  }
+
+  async getAvailablePermissionGroupsForOrganization(
+    organization: Organization,
+  ): Promise<Array<PermissionGroup>> {
+    const filter = this.createFeatureFlagPermissionFilter(organization);
+
+    return OrganizationAccessTokensPermissions.permissionGroups
+      .map(group => ({
+        ...group,
+        permissions: group.permissions
+          .filter(permission => filter(permission.id))
+          .map(permission => ({
+            ...permission,
+            isAssignableByViewer: true,
+          })),
+      }))
+      .filter(group => group.permissions.length !== 0);
+  }
+
+  async getAvailablePermissionGroupsForMembership(membership: OrganizationMembership) {
```

Comment:
`getAvailablePermissionGroupsForMembership`, `getAvailablePermissionGroupsForOrganization` and `getAvailablePermissionsGroupsForProject` are now the "source" of truth for the assignable permissions on the frontend.

All the permissions are filtered on:
- feature flag
- resource level (org or project)
- membership permissions (only for PAT)